vertx-pojongo/src/main/java/de/braintags/vertx/jomnigate/mongo/MongoUtil.java

```diff
@@ -210,22 +211,27 @@ private static void addIndexOptions(final JsonObject indexDef,
       futures.add(future);
 
       switch (option.getFeature()) {
-      case UNIQUE:
-        indexDef.put("unique", value);
-        future.complete();
-        break;
-      case PARTIAL_FILTER_EXPRESSION:
-        convertFilterExpression((String) value, indexDef, mapper, dataStore, future);
-        break;
-      default:
-        future.fail(new IllegalArgumentException("Unknown IndexFeature: " + option.getFeature()));
+        case UNIQUE:
+          indexDef.put("unique", value);
+          future.complete();
+          break;
+        case PARTIAL_FILTER_EXPRESSION:
+          convertFilterExpression((String) value, indexDef, mapper, dataStore, future);
+          break;
+        case SPARSE:
+          indexDef.put("sparse", value);
+          future.complete();
+          break;
+        default:
+          future.fail(new IllegalArgumentException("Unknown IndexFeature: " + option.getFeature()));
       }
     }
     CompositeFuture.all(futures).setHandler(result -> handler.handle(result.map(indexDef)));
   }
 
-  private static void convertFilterExpression(String filterExpression, final JsonObject indexDef, IMapper<?> mapper,
-      final MongoDataStore dataStore, Future<Void> future) {
+  private static void convertFilterExpression(final String filterExpression, final JsonObject indexDef,
```

Comment:
please check your formatting options, auto format, cause linebreaks should be different here